apps/appsets/components.yaml

```diff
@@ -18,60 +18,84 @@ spec:
           - list:
               elements:
                 - component: dex
+                  source:
+                    repoURL: '{{index .metadata.annotations "uc_repo_git_url"}}'
+                    targetRevision: '{{index .metadata.annotations "uc_repo_ref"}}'
+                    path: 'components/dex'
                 - component: openstack
+                  source:
+                    repoURL: '{{index .metadata.annotations "uc_repo_git_url"}}'
```

Comment:
this section is quite repetitive. What do you think about using YAML aliases and merging here?

```yaml
- component: openstack
  source: &uc_component
    repoURL: '{{index .metadata.annotations "uc_repo_git_url"}}'
    targetRevision: '{{index .metadata.annotations "uc_repo_ref"}}'
    path: 'components/openstack'
- component: undersync
  source:
    <<: *uc_component
    path: 'components/undersync'
- component: nautobot
  source:
    <<: *uc_component
    path: 'components/nautobot'
- component: argo
  source:
    <<: *uc_component
    path: 'components/argo'
    kustomize:
      patches:
```

Reply:
Good suggestion! It works.